routes/index.js

```diff
@@ -50,42 +53,59 @@ router.post('/api/users', function (req, res) {
     console.log('Input field empty')
     return res.status(400).send('Input field empty')
   }
-
   // Checks for duplicate
-  const checkdup = {
-    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
-    values: [email.toLowerCase()]
-  }
-  db.query(checkdup)
+  CheckDup(email)
     .then(result => {
-      if (result.rows[0].count !== 0) {
-        res.status(240)
+      if (result != 0) {
+        // console.log(result)
+        res.status(409).send({message:"exist"})
+      } else {
+
+        // Insert user data
+        const query = {
+          text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
+          values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret]
+        }
+        db.query(query)
+          .then(result => {
+            res.send([firstname, lastname, email, secret])
+          })
+          .catch(err => {
+            res.status(500).json(err)
+          })
       }
     })
+ 
+  
+})
 
-  // Insert user data
-  const query = {
-    text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
-    values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret]
+// Helper function for checking user duplicates using email
+// Goes through database and counts where email matches
+async function CheckDup(email) {
+  const checkdup = {
+    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
+    values: [email.toLowerCase()]
   }
-  db.query(query)
-    .then(result => {
-      res.send([firstname, lastname, email, secret])
-    })
-    .catch(err => {
-      res.status(400).json(err)
-    })
-})
+  return await db.query(checkdup)
+  .then(function(result) {
+    return result.rows[0].count
```

Comment:
There's no point in using`.then` on an `await`ed value. That is, this should say
```
const result = await db.query(checkdup)
return result.rows[0].count
```

Also: this should probably return a boolean, rather than requiring the callee to parse the count (unless you think it might be useful to have the count). If you return a boolean tho, make sure the name of the function is clear what true represents (eg `emailAlreadyRegistered` vs `CheckDup`, which is ambiguous)